### DISCUSSION TITLE
Need information on User Update API

### DISCUSSION THREAD

**Discussion by Xitija** [Sunbird-Lern/Community#67] [Query]
Hello All,

- I want to understand the difference between 
   /user/v1/update       /user/v2/update          /user/v3/update 

Along with that I want to know more on 
- For updating the user Sunbird uses the api which starts with `learner` but api in the [documentation](https://lern.sunbird.org/learn/product-and-developer-guide/user-and-org-service/api-documentation/user-management) starts with `api`
for example : `learner/user/v3/update` vs `api/user/v3/update`
Can anyone please tell me what is the difference between them

**Comment by reshmi-nair** [comment]
@Xitija Mainly the api requests differ on the usertype and location data format as below:
/user/v1/update
{
 "request": {
   "userId": "48382e16-282c-4eec-854f-c8112e4aefba",
  "userType":"teacher",
 "userSubType":"HT",
"locationCodes":[]
 }
}
/user/v2/update
{
 "request": {
   "userId": "48382e16-282c-4eec-854f-c8112e4aefba",
  "profileUserType":{"type":"teacher","subType":"HT"},
  "profileLocation": [{"code": "0912","type": "district"}]
 }
}
/user/v3/update
{
 "request": {
   "userId": "48382e16-282c-4eec-854f-c8112e4aefba",
  "profileUserTypes":[{"subType":["hm1"],"type":"administrator1"},{"subType":["hm2"],"type":"administrator2"}]
 }
}

API documentation says how to call the APIs directly from postman, so we can call using /api because thats what the kong layer understands. /learner is used by Sunbird-Ed code, this means it is a userorg API. But internally /learner is replaced with /api before making the actual API call.

**Reply by Xitija** [reply]
That means the API called is same but thorugh different ways.
But headers of both API calls are different so` learner` takes session_id and cookie - connect.sid
and `api` takes Authorization and x-authenticated-user-token . 
Is Authorization and x-authenticated-user-token added during the API call internally by Sunbird?
